microprofile/lra/jax-rs/src/main/java/io/helidon/microprofile/lra/HandlerService.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.helidon.microprofile.lra;
+
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import io.helidon.lra.coordinator.client.CoordinatorClient;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.lra.annotation.AfterLRA;
+import org.eclipse.microprofile.lra.annotation.Complete;
+import org.eclipse.microprofile.lra.annotation.Status;
+import org.eclipse.microprofile.lra.annotation.ws.rs.LRA;
+import org.eclipse.microprofile.lra.annotation.ws.rs.Leave;
+import org.jboss.jandex.AnnotationInstance;
+
+@ApplicationScoped
+class HandlerService {
+
+    @Inject
+    private CoordinatorClient coordinatorClient;
+
+    @Inject
+    private InspectionService inspectionService;
+
+    @Inject
+    private ParticipantService participantService;
+
+    @ConfigProperty(name = "mp.lra.propagation.active", defaultValue = "true")
+    private boolean propagate;
+
+    private static final Map<String, AnnotationHandler.HandlerMaker> HANDLER_SUPPLIERS =
```

Comment:
constants must be defined before fields.
Also please use constructor injection

Reply:
fixed